src/vm/assembly.cpp

```diff
@@ -2722,7 +2722,7 @@ INT32 Assembly::ExecuteMainMethod(PTRARRAYREF *stringArgs)
     //to decide when the process should get torn down.  So, don't call it from
     // AppDomain.ExecuteAssembly()
     if (pMeth) {
-        if (stringArgs == NULL)
+        if (waitForOtherThreads)
```

Comment:
To preserved existing behavior 
waitForOtherThreads needs to be TRUE when (stringArgs == NULL)